code/src/components/app-dropdown/AppDropdown.tsx

```diff
@@ -0,0 +1,84 @@
+import { useState, useRef, MouseEvent, FocusEvent, ReactNode } from "react";
+import KeyboardArrowDownIcon from "@mui/icons-material/KeyboardArrowDown";
+import KeyboardArrowUpIcon from "@mui/icons-material/KeyboardArrowUp";
+import AppBox from "@/components/app-box/AppBox";
+import AppTypography from "@/components/app-typography/AppTypography";
+import AppDropdownOptions from "@/components/app-dropdown/components/AppDropdownOptions";
+import { AppDropdownProps } from "@/components/app-dropdown/AppDropdown.types";
+
+import "@/components/app-dropdown/AppDropdown.scss";
+
+const AppDropdown = ({
+  options,
+  onSelect,
+  defaultLabel,
+  className
+}: AppDropdownProps) => {
```

Comment:
If we create reusable component, we should use all possible props, like `...props`

Reply:
If you defined reusable component, it should have a lot of possible use cases. If it is not planned for you, move it to layouts please

Reply:
> If you defined reusable component, it should have a lot of possible use cases. If it is not planned for you, move it to layouts please

Are you sure to move it to layouts and where this layout should be? Why ? A lof of possible use cases it's about `AppTypography` variants and some others. It's component no more. You suggest to change different color or some border color for it?) Sure it's possible just pass your `className` as a prop. It's reusable component not layout when you can pass your own styles and rewrite exists, pass the default value.. You can check again the props)

Reply:
Yeap, but in my opinion it is better to spread all props if we need them in some point of time if you want to make component trully reusable and customisable. You can check already existing app components as a reference. I think we can create `containers` folder in the future as on s2s project. `layouts` folder will contain header, footer, navbar, sidebar and different wrappers and boxes and `containers` folder is perfect for components which you introduced. For now it is not an urgent task, but it is worth implementing in the future